tests/test_dao.py

```diff
@@ -0,0 +1,522 @@
+    assert response.status_code == 200
+    unsigned_transaction = response.json()
+    assert unsigned_transaction['transaction']
+    assert len(unsigned_transaction['signatures']) == 0
+
+    response = requests.post(NODE_URL+'/sign-transaction', json={
+        "wallet_data": WALLET,
+        "transaction_data": unsigned_transaction
+    })
+
+    assert response.status_code == 200
+    signed_transaction = response.json()
+    assert signed_transaction['transaction']
+    assert signed_transaction['signatures']
+    assert len(signed_transaction['signatures']) == 1
+
+    response = requests.post(
+        NODE_URL+'/validate-transaction', json=signed_transaction)
+    assert response.status_code == 200
+
+    if TEST_ENV == 'local':
+        response = requests.post(
+            NODE_URL + '/run-updater?add_to_chain_before_consensus=true')
+    else:
+        print('Waiting to mine block')
+        time.sleep(BLOCK_WAIT_TIME)
+
+    response = requests.get(NODE_URL+'/download-state')
+    assert response.status_code == 200
+    state = response.json()
+
+    wallets = state['wallets']
+    wallet_in_state = next(
+        x for x in wallets if x['wallet_address'] == wallet['address'])
+    assert wallet_in_state
+
+    print("Wallet created with address "+wallet['address'])
+    print('Test passed.')
+    return wallet
+
+
+def create_token(wallet, owner, token_name, token_code, amount):
+    response = requests.post(NODE_URL+'/add-token', json={
+        "token_name": token_name,
+        "token_code": token_code,
+        "token_type": "1",
+        "first_owner": owner,
+        "custodian": wallet["address"],
+        "legal_doc": "686f72957d4da564e405923d5ce8311b6567cedca434d252888cb566a5b4c401",
+        "amount_created": amount,
+        "tokendecimal": 0,
+        "disallowed_regions": [],
+        "is_smart_contract_token": False,
+        "token_attributes": {}
+    })
+
+    assert response.status_code == 200
+    unsigned_transaction = response.json()
+    assert unsigned_transaction['transaction']
+    assert len(unsigned_transaction['signatures']) == 0
+
+    response = requests.post(NODE_URL+'/sign-transaction', json={
+        "wallet_data": wallet,
+        "transaction_data": unsigned_transaction
+    })
+
+    print("adding token")
+    assert response.status_code == 200
+    signed_transaction = response.json()
+    print("signing tx")
+    assert signed_transaction['transaction']
+    assert signed_transaction['signatures']
+    assert len(signed_transaction['signatures']) == 1
+
+    print("validating tx")
+    response = requests.post(
+        NODE_URL+'/validate-transaction', json=signed_transaction)
+    assert response.status_code == 200
+
+    if TEST_ENV == 'local':
+        response = requests.post(
+            NODE_URL + '/run-updater?add_to_chain_before_consensus=true')
+    else:
+        print('Waiting to mine block')
+        time.sleep(BLOCK_WAIT_TIME)
+
+
+wallet_founder1 = create_wallet()
```

Comment:
@Prannav-k let's not call this function here. this will cause vscode to call this everytime tests are discovered. instead use a setup method which can be configured to be called before running tests in pytest.

Reply:
@kousthubraja yes, it is now solved with using mixture of python fixtures and request chain (to use results of previous tests). Below is sample
```
@pytest.fixture
def dao_details():
    dao_details = {
        "dao_name":"sam",
    }

    return dao_details


@pytest.mark.dependency()
def test_t1(dao_details,request):
    sam = dao_details["sam"]
    print(sam)
    dao_details["dao_address"] = "dao_address"
    request.config.cache.set('dao_address', 'dao_address')


@pytest.mark.dependency(depends=['test_t1'])
def test_t2(request):
    vam = request.config.cache.get('dao_address', None)
    print(vam)
```